### ISSUE TITLE
Client Credentials grant breaks when client secret contains %

### ISSUE THREAD

**Issue by adamrbennett** [panva/node-oidc-provider#267]
With the recent changes in e691a105fef72a7433e4bfce11fd3a55ac2abaef, the Client Credentials grant now fails if the Client Secret contains a `%`. I haven't tested, but I imagine this also applies to Client IDs, as well as other URL "unsafe" characters.

1. Run latest oidc-provider
1. Perform the Client Credentials grant with a secret containing a `%`

```
oidc-provider:server_error uuid=75eeb1ad-36f9-48fc-ae13-07f3b95dd5d7 path=/token method=POST error=URIError: URI malformed
at decodeURIComponent (<anonymous>)
at decodeAuthToken (/opt/app/node_modules/oidc-provider/lib/shared/find_client_id.js:6:10)
at findClientId (/opt/app/node_modules/oidc-provider/lib/shared/find_client_id.js:29:43)
at dispatch (/opt/app/node_modules/oidc-provider/node_modules/koa-compose/index.js:42:32)
at next (/opt/app/node_modules/oidc-provider/node_modules/koa-compose/index.js:43:18)
at checkDupes (/opt/app/node_modules/oidc-provider/lib/shared/check_dupes.js:12:9)
at dispatch (/opt/app/node_modules/oidc-provider/node_modules/koa-compose/index.js:42:32)
at next (/opt/app/node_modules/oidc-provider/node_modules/koa-compose/index.js:43:18)
at assembleParams (/opt/app/node_modules/oidc-provider/lib/shared/get_params.js:21:11)
at dispatch (/opt/app/node_modules/oidc-provider/node_modules/koa-compose/index.js:42:32)
```

**Comment by panva** [comment]
Are you properly encoding those characters?

I agree the server_error response needs fixing, no problem, but either way the client authentication would fail with conform behavior in now.

**Comment by panva** [comment]
Thinking a bit altho this was a bug fix, it is big enough of a breaking change that it deserves a major version bump and a proper thick bold explanation in the changelog.

**Comment by panva** [comment]
1) deprecated 2.18.1
2) published 2.18.0 again as 2.18.2
3) release 3.0.0 with the fix applied and instead of a server_error properly communicate invalid request due to the credentials not being properly encoded.
